source/develop/release-management/features/storage/cinderlib-integration.html.md

```diff
@@ -0,0 +1,145 @@
+
+#### Create Disk
+
+The user creates a disk on the storage domain:
+  - The Engine calls a Cinderlib API with the driver backend parameters, the volume id, the volume size and the DB URL.
+  - If the call to Cinderlib API is successful, the Engine creates the Disk in the DB.
+
+#### Delete Disk
+
+The user delete a disk on the storage domain:
+  - The Engine calls a Cinderlib API with the driver backend parameters, the volume id, the volume size and the DB URL to delete the volume.
+  - If the call to Cinderlib API is successful, the Engine deletes the disk in the DB
+
+### Connection information from Vdsm host
+
+In order to be able to expose the volume to a host, Cinderlib needs to have its connection information.
+
+As part of GetCapabilities Verb a new entry will be added to provide the needed information.
+
+The Vdsm uses the Cinder's 'os_brick' library to collect the data.
+
+#### Run VM
+
+The user runs a VM with a Cinderlib disk :
+  - The Engine calls a Cinderlib API to connect the volume using the connection information for the specific host and return the needed data for the attachment.
+  - The Engine calls a Vdsm verb to attach the volume with the data needed. The Vdsm stores locally the attachment details that will be needed for detach, and then return the path.
+  - The Engine uses the returned path for the volume in the VM XML.
+
+#### Run VM
+
+The user runs a VM with a Cinderlib disk :
+  - The Engine calls a Cinderlib API to connect the volume using the connection information for the specific host and return the needed data for the attachment.
+  - The Engine calls a Vdsm verb to attach the volume with the data needed. The Vdsm stores locally the attachment details that will be needed for detach, and then return the path.
+  - The Engine uses the returned path for the volume in the VM XML.
+
+
+#### Stop VM
+
+The user stops a VM with a Cinderlib disk :
+  - The Engine calls a Vdsm verb to detach the volume with the data needed. The Vdsm use the attachment details stored locally.
+  - The Engine calls a Cinderlib API to disconnect the volume.
+
+
+### VDSM
+
+- Add connection information as part of GetCapabilities verb using 'os_brick'.
+- Add an 'attach' verb that will get the details of the volume, attach to the local host, persist the data for attachment and return the path.
+
+// TODO describe additional changes needed
+
+### Engine
+
+- Add DB support for additional Storage Domain information.
+- Block not supported storage operations.
+- Python executable giving API to Cinderlib
+
+// TODO describe additional changes needed
+
+### Installation
+
+Cinderlib will be packaged in RPM.
+The Cinder drivers and Cinderlib needs to be installed in the Engine.
+OS-Brick needs to installed on the hosts
+
+
+### Open Issues / Limitations
+
+- Cinderlib needs to be packaged in RPM that will be required by the Engine.
+- os-brick needs to be installed on the hosts.
```

Comment:
maybe worth to mention that right now for installing cinderlib you'll need to install the whole openstack-cinder service?

Reply:
Done